### ISSUE TITLE
[Accessibility] Buttons incorrectly given the role of links

### ISSUE THREAD

**Issue by amberhinds** [woocommerce/woocommerce-gateway-amazon-pay#335]
## Audit Information
This issue came from an audit WooCommerce had Equalize Digital complete in December 2025. I'm moving the issue out of Linear (WooA11Y-423) to this repo.

- EQD Audit ID: WooExt-519
- Component/Page: Checkout
- Affected Element: Select button
- URL where issue was discovered: https://apay-woo.mystagingwebsite.com/checkout/

## WCAG Compliance

- Success Criterion: 4.1.2 Name, Role, Value
- WCAG Version: 2
- Level: A
- Severity: Medium

## Issue Description

Buttons incorrectly have assigned the `role="link"` attribute or `<a>` tags on them, though they do not function as links and are not redirecting to another URL. Buttons are elements that trigger an action on a page (opening a lightbox, expanding an accordion, submitting a form).

### Impacted Populations

Screen reader users (blind/low vision and neurodivergent people)

## Recommended Fix

Use the appropriate `<button>` tag on the elements.

If you cannot replace a `<a>` tag with a button, it is possible to remediate a link to function as a button with ARIA. The link would need to be given a button role and `href="#"` to ensure that it can be focused with the keyboard. You would also need to write JavaScript to ensure that the link will function properly as a button for keyboard users and can be triggered with the space bar in addition to the return/enter key. For more information on turning a link into a button, see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

## Code Example

`<a href="#" class="wc-apa-widget-change amazonpay-merchant-shadow-root-parent-element-for-executing-modal-script" id="amazon_change_payment_method" data-wc_apa_change_bind="changePayment">Select</a>`

## Screenshots

https://drive.google.com/file/d/1aY4u_JhXWz5PEXbiUm4OBqxyNkqZrUVx/view?usp=drivesdk
